### PR TITLE
Register some event listeners

### LINKED ISSUE #207421: Leaking event listener
https://github.com/microsoft/vscode/blob/be776fbfff16d18b19457cc271e4e049b766490d/src/vs/workbench/contrib/terminal/browser/baseTerminalBackend.ts#L49

### LINKED ISSUE #207430: Leaking event listener
https://github.com/microsoft/vscode/blob/be776fbfff16d18b19457cc271e4e049b766490d/src/vs/workbench/contrib/terminal/electron-sandbox/localTerminalBackend.ts#L98

### LINKED ISSUE #207435: Leaking event listener
https://github.com/microsoft/vscode/blob/be776fbfff16d18b19457cc271e4e049b766490d/src/vs/workbench/contrib/terminal/browser/terminalGroupService.ts#L73

### LINKED ISSUE #207440: Leaking event listener
https://github.com/microsoft/vscode/blob/be776fbfff16d18b19457cc271e4e049b766490d/src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts#L92

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #207430
Fixes #207435
Fixes #207421
Fixes https://github.com/microsoft/vscode/issues/207440

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 1]
```diff
@@ -520,14 +520,14 @@ export class TerminalService extends Disposable implements ITerminalService {
 	}
 
 	private _attachProcessLayoutListeners(): void {
```
Comment (jrieken) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:522]: This looks like a method that's called repeatedly for the lifecycle of this service. If that's true `_register` must not be used, instead a session disposable that clears whatever a previous invocation has store should be used

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:522]: This is actually only called once here:

https://github.com/microsoft/vscode/blob/f933441a9e17555bb06fa98a29c199e559b4bf9b/src/vs/workbench/contrib/terminal/browser/terminalService.ts#L309-L313

The reconnect happens once per session and only for the "primary" backend, ie. you cannot reconnect to local terminals in a remote workspace.

Comment (jrieken) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:522]: You know best, but `initializePrimaryBackend` is a public method and callable by anyone

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:522]: Made it private 👍

**Review by jrieken — APPROVED** [review]

**Pull request merged**